### ISSUE TITLE
Support `aarch64-linux`

### ISSUE THREAD

**Issue by mcginty** [tadfisher/android-nixpkgs#50]
This is particularly useful when developing with VM setups like what [`lima`](https://github.com/lima-vm/lima) offers by default.

Not sure what this will take exactly, and am relatively new to nix, so sorry for not being more useful in providing a PR!

**Comment by tadfisher** [comment]
This project just repackages what AOSP provides, and they don't provide AArch64 Linux builds for any SDK package.

**Comment by initialed85** [comment]
> This is particularly useful when developing with VM setups like what [`lima`](https://github.com/lima-vm/lima) offers by default.
> 
> Not sure what this will take exactly, and am relatively new to nix, so sorry for not being more useful in providing a PR!

@mcginty - I'm not sure if you're still looking for this- I tried to make a start; ref.: https://github.com/tadfisher/android-nixpkgs/compare/main...initialed85:android-nixpkgs:main

I'm using [cachix/devenv](https://github.com/cachix/devenv) and so the files of relevance for me `devenv.yaml` (which I've got pointed at my fork):

```yaml
inputs:

  nixpkgs:
    url: github:NixOS/nixpkgs/nixpkgs-unstable

  nixpkgs-stable:
    url: github:NixOS/nixpkgs/23.05

  android-nixpkgs:
    # url: github:tadfisher/android-nixpkgs/main
    url: github:initialed85/android-nixpkgs/main
    inputs:
      nixpkgs:
        follows: nixpkgs
```

And `devenv.nix`: 

```nix
{ inputs, pkgs, lib, config, ... }:

let
  nixpkgs = import inputs.nixpkgs {
    system = pkgs.stdenv.system;
  };

  android-nixpkgs = import inputs.android-nixpkgs {
    channel = "stable";
  };

  # note: these are coupled to the avdmanager invocation in run-emulator.sh
  android-sdk = android-nixpkgs.sdk (sdkPkgs: with sdkPkgs; [
    cmdline-tools-latest
    build-tools-32-0-0
    # platform-tools
    platforms-android-31
    emulator
    system-images-android-31-google-apis-arm64-v8a
    system-images-android-31-google-apis-x86-64
    ndk-25-2-9519653
  ]);
in
{
  env.ANDROID_HOME = "${android-sdk}/share/android-sdk";
  env.ANDROID_SDK_ROOT = "${android-sdk}/share/android-sdk";
  env.ANDROID_NDK_HOME = "${android-sdk}/share/android-sdk/ndk/25.2.9519653";

  packages = [
    pkgs.jdk11
    android-sdk
    pkgs.android-tools
  ];

  enterShell = ''
    set -e

    ANDROID_USER_HOME=$(pwd)/.android
    ANDROID_AVD_HOME=$(pwd)/.android/avd

    export ANDROID_USER_HOME
    export ANDROID_AVD_HOME

    test -e "$ANDROID_USER_HOME" || mkdir -p "$ANDROID_USER_HOME"
    test -e "$ANDROID_AVD_HOME" || mkdir -p "$ANDROID_AVD_HOME"

    set +e
  '';
}
```

You'll note that I had `platform-tools` commented and then separately installed `pkgs.android-tools`- this is because `platform-tools` from `tadfisher/android-nixpkgs` refers to the official Linux release (which seems to only support x86-64, so `adb` didn't work); while changing to [`pkgs.android-tools`](https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/tools/misc/android-tools/default.nix#L31) does make `adb` work it breaks other things that `tadfisher/android-nixpkgs` sets up (basically the `avdmanager` gets unhappy stating it can't find `platform-tools`).

I was able to confirm that by hacking the various executables into the expected `platform-tools/bin` path I could at least get `avdmanager` to work again, however then I got stuck getting the emulator to actually run.

I think it was mostly because I was testing on an M2 MacBook in an Ubuntu Linux VM using VMware Fusion and I think it doesn't support nested virtualisation (so I can't do KVM).

So basically, if you have a real AArch64 Linux machine at your disposal (e.g. an Apple Silicon Mac running native Linux) then you _might_ be able to get it to work if you can figure out how to merge [nixpkgs/android-tools](https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/tools/misc/android-tools/default.nix#L31) and my hacks to `tadfisher/android-nixpkgs`.

You might already be across it- I found this dude's gist to be a good reference: https://gist.github.com/atyachin/2f7c6054c4cd6945397165a23623987d
